javascript/modules/client/src/internal/interfaces/common.ts

```diff
@@ -41,3 +41,40 @@ export type GasFeeEstimation = {
   average: bigint;
   max: bigint;
 };
+
+/**
+ * Contains the base structure of a proposal
+ */
+export interface IProposal {
```

Comment:
- IMO this should be a strict type, not a flexible interface
- We should use it in the code
- Let's make to mirror the fields defined here correspond to the the logic above